app/src/org/odk/collect/android/views/media/AudioButton.java

```diff
@@ -362,24 +278,83 @@ private String formatTime(Integer milliseconds) {
         return returnValue;
     }
 
-    public void startPlaying() {
-        logAction("start");
-        controller.playCurrentMediaEntity();
-        setStateToPlaying();
-    }
+    public AudioController buildAudioControllerInstance() {
+        return new AudioController() {
+            private MediaPlayer mp;
+            boolean alive = false;
 
-    public void endPlaying() {
-        logAction("stop");
-        if (!currentState.equals(MediaState.Playing)) {
-            controller.releaseCurrentMediaEntity();
-            setStateToReady();
-        }
-    }
+            @Override
+            public MediaEntity getCurrMedia() {
+                return null;
+            }
 
-    public void pausePlaying() {
-        logAction("pause");
-        controller.pauseCurrentMediaEntity();
-        setStateToPaused();
-    }
+            @Override
+            public void setCurrent(MediaEntity newEntity) {
+                this.mp = newEntity.getPlayer();
+            }
+
+            @Override
+            public void setCurrent(MediaEntity newEntity, AudioButton newButton) {
+                setCurrent(newEntity);
+            }
+
+            @Override
+            public void releaseCurrentMediaEntity() {
+                if (mp != null) {
+                    mp.reset();
+                    mp.release();
+                    mp = null;
+                    alive = false;
+                }
+            }
+
+            @Override
+            public Object getMediaEntityId() {
+                return residingViewId;
+            }
+
+            @Override
+            public void refreshCurrentAudioButton(AudioButton clicked) { }
+
+            @Override
+            public void saveEntityStateAndClear() { }
+
+            @Override
+            public void setMediaEntityState(MediaState state) { }
+
+            @Override
+            public void playCurrentMediaEntity() {
+                alive = true;
+                mp.start();
+            }
+
+            @Override
+            public void pauseCurrentMediaEntity() { }
+
+            @Override
+            public void setCurrentAudioButton(AudioButton b) { }
+
+            @Override
+            public void removeCurrentMediaEntity() { }
+
+            @Override
+            public void attemptSetStateToPauseForRenewal() { }
```

Comment:
No, this is an audio controller interface that Aliza created. It looks like she had plans to include more functionality, but has for the moment only implemented the non-empty methods in this anonymous class. These line additions are just me moving her code into a helper function.

Looks like I can't take out the empty overrides because the interface doesn't have any default implementations. Later I will probably refactor this further to do such things.